MyLibrary/Library/Controllers/BookController.cs

```diff
@@ -0,0 +1,342 @@
+        /// </summary>
+        /// <param name="bookId">The book identifier</param>
+        /// <param name="genreId">The genre identifier</param>
+        /// <returns>
+        /// HTTP status code Created() with id of new record in pairsBookGenre
+        /// or HTTP status code BadRequest() if the book with such id doesn`t exist
+        /// or HTTP status code BadRequest() if the genre with such id doesn`t exist
+        /// </returns>
+        [HttpPost("bookId={bookId}&genreId={genreId}")]
+        public IActionResult AddGenreToBook(int bookId, int genreId)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest("Invalid input");
+            }
+
+            try
+            {
+                int newPairBookGenreId = this.libraryService.AddGenreToBook(bookId, genreId);
+                return Created("PairBookAuthor", newPairBookGenreId);
+            }
+            catch (ArgumentException exception)
+            {
+                return BadRequest(exception.Message);
+            }
+        }
+
+        // DELETE: api/book/
+        /// <summary>
+        /// Removes the genre from book
+        /// </summary>
+        /// <param name="bookId">The book identifier</param>
+        /// <param name="genreId">The genre identifier</param>
+        /// <returns>
+        /// HTTP status code Ok() with removedPairBookGenreId
+        /// or HTTP status code NotFound() if there are o specified genre in specified book
+        /// or HTTP status code BadRequest() if there is no author or book with specified ids
+        /// </returns>
+        [HttpDelete("bookId={bookId}&genreId={genreId}")]
+        public IActionResult RemoveGenreFromBook(int bookId, int genreId)
+        {
+            try
+            {
+                int? removedPairBookGenreId = this.libraryService.RemoveGenreFromBook(bookId, genreId);
+                if (removedPairBookGenreId == null)
+                {
+                    return NotFound("No specified genre in specified book!");
+                }
+
+                return Ok(removedPairBookGenreId);
+            }
+            catch (ArgumentException exception)
+            {
+                return BadRequest(exception.Message);
+            }
+        }
+
+        // GET: api/book/
+        /// <summary>
+        /// Get all books by specified genre
+        /// </summary>
+        /// <param name="genreId">The genre identifier</param>
+        /// <returns>
+        /// HTTP status code Ok() with list of found books
+        /// or HTTP status code NotFound() if there are no books with such genre
+        /// </returns>
+        [HttpGet("genreId={genreId}")]
+        public IActionResult GetBooksByGenre(int genreId)
+        {
+            try
+            {
+                List<Book> foundBooks = new List<Book>();
+                foundBooks = this.libraryService.GetBooksByGenre(genreId).ToList();
+
+                if (foundBooks.Count == 0)
```

Comment:
https://stackoverflow.com/questions/305092/which-method-performs-better-any-vs-count-0

in your case with any, you wouldn't need to call .ToList() as I see